Packs/Office365AndAzureAuditLog/Integrations/MicrosoftPolicyAndComplianceAuditLog/MicrosoftPolicyAndComplianceAuditLog.ps1

```diff
@@ -26,36 +26,26 @@ function UpdateIntegrationContext([OAuth2DeviceCodeClient]$client){
 }
 
 function CreateNewSession {
-    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidUsingConvertToSecureStringWithPlainText', '', Scope='Function')]
-    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidUsingPlainTextForPassword', '', Scope='Function')]
-    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSReviewUnusedParameter', '', Scope='Function')]
-    param([string]$url, [string]$upn, [string]$password, [string]$bearer_token, [bool]$insecure, [bool]$proxy)
-    if ($password){
-        $url = "$url/powershell-liveid"
-    }
-    else
-    {
-        $url = "$url/powershell-liveid?BasicAuthToOAuthConversion=true"
-    }
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidUsingConvertToSecureStringWithPlainText', '', Scope = 'Function')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidUsingPlainTextForPassword', '', Scope = 'Function')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSReviewUnusedParameter', '', Scope = 'Function')]
+    param([string]$url, [string]$upn, [string]$bearer_token, [bool]$insecure, [bool]$proxy)
+    $url = "$url/powershell-liveid?BasicAuthToOAuthConversion=true"
+    $credential = ConvertTo-SecureString "Bearer $bearer_token" -AsPlainText -Force
```

Comment:
So this will still technically use _client_ basic auth and it will require customers to enable client basic auth on their service account. This was tested in our environment?